### DISCUSSION TITLE
iPhone tracking when screen off

### DISCUSSION THREAD

**Discussion by harphere** [ESPresense/ESPresense#492] [Q&A] [answered]
I find when the iPhone 12 Pro screen goes off it stops broadcasting the device id. I understand this could be privacy related on Apple's part but not really helpful when using it to keep the lights on in a room. Anyone else face the same thing? Any workaround?

**Comment by martindell** [comment]
Yep, exactly the same problem for me here.  Every app I tried (apple & android) stops sending a beacon when the screen is off.  The only exception I found was the UK's NHS Covid 19 app which reliably sends a beacon whatever the phone is doing.

I thought I'd found the answer with that because I just want to know if 'anyone' is in the room so it doesn't matter to me that the beacon ID's between phones are not unique.  However, the iphone BT signal is so strong that the base station thinks it's always 'in the room' compared to my android BT signal which is comparatively weak and works as I'd expect it to.

I'm using a M5StickC Plus as the base station and have run out of beacons to try - I'm definitely not carrying a Tile around with me just for this.  

I tried my Oura ring as a beacon but it has the opposite problem - BT signal too weak to detect me further than 2m from the base station: https://www.reddit.com/r/ouraring/comments/qj3pqv/i_use_my_oura_ring_a_little_differently/

**Comment by harphere** [comment]
Hmmm. I was hoping BLE IRK would help to fix this dilemma but no luck.

**Reply by DTTerastar** [reply]
My iPhone blasts out nearby info every 500ms no matter what.  It even does some ble when turned off in the later iOS versions.  Not sure why you're seeing something different.

**Reply by harphere** [reply]
Curious to know if your Auto Lock under Display setting is set to 'Never'? In my case, the only time I get a constant stream of data is when the screen is always awake but that leads to obvious quicker battery drain. When I set any Display timeout is when the phone stops transmitting after hitting the away_timeout defined under mqtt_room.

**Comment by DTTerastar** [comment]
https://github.com/ESPresense/ESPresense/issues/574 mentions the same issue.  See if you 2 have anything in common perhaps?

**Comment by 51av0sh** [comment]
Just wanted to add myself to this. My wife's iPhone has the same issue. Tracking completely stops when screen goes off. I just tried pairing her phone to one of the ESP32s using the latest beta and using the irk instead but that didn't solve it.

**Comment by LynnOnTheWeb** [comment]
I have the same issue. After the initial pairing, the tracking stops. Checking the MQTT logs it shows that the connection is lost and it never reconnects. IPhone 13 Pro, running iOS 15.6

**Reply by 51av0sh** [reply]
Same here. After I paired my wife's phone and switched to using the irk, tracking has stopped working (even with screen on). I tried even switching back to the apple* id and can't get that to work anymore. I don't see anything in the esp logs.

**Comment by 51av0sh** [comment]
A curious thing I've noticed is that since the latest beta (b5) after pairing, apple devices are no longer being sent to my mqtt queue. I do see them in the esp logs but none show up in mqtt

**Reply by DTTerastar** [reply]
The irk is a better id for the mac, so it's used instead of apple:

**Reply by 51av0sh** [reply]
Makes sense. My problem is that since I paired her phone and started using ble irk now her phone stops tracking after 10-15 seconds even with the screen on. I tried going back to using the apple* id but now I can't use it anymore since it's not being sent to mqtt. I unpaired her phone and removed the irk number from espresense/settings and I'm still not able to get her phone to track anymore. Do you have any suggestions? I'd like to at least go back to the way it was before (using the apple* id since that was working fine while screen was on).

**Reply by 51av0sh** [reply]
To clarify. When I say it stops tracking I mean that the room and distance no longer update and looking at the mqtt topic for her phone I don't see any new messages

**Reply by 51av0sh** [reply]
No clue why but it started working again and now it's updating as usual (but now using irk). I haven't changed anything so I have no idea why/how but it's working so I'm just gonna let it be

**Comment by 51av0sh** [comment]
Sorry for the spam but just wanted to provide more info in case it helped. Not sure if it's just coincidence but I noticed that as soon as the esp looks for an update, my wife's phone starts tracking for about 10-15 seconds and then it stops again. I tried rebooting the esp closest to her phone to see if that did anything but no dice.  The log below shows when the last check for update happened and her phone suddenly started tracking again but stopped shortly after. 

![Screenshot 2022-08-13 114601](https://user-images.githubusercontent.com/85576728/184506858-43b47033-902b-4629-ae6b-b6f59ff77477.png)

......

Update: Go figure. After a few hours and nothing being different, my wife's iPhone is updating again as normal. Strange

**Reply by DTTerastar** [reply]
I figured this out and it's now fixed in the latest release.

**Reply by 51av0sh** [reply]
Can confirm this latest build works perfectly. My wife is not home so I haven't tested her iPhone13 but I just woke up her apple watch 6 that she left here and it's been tracking for the past 10 minutes (using the apple* id). Solid and no drops even with the screen off. Fantastic work my guy.

**Reply by 51av0sh** [reply]
Looks like it stopped tracking exactly at the 20 min mark. Still way longer than I've ever seen it track (at least her watch). I'll see how her phone does when she's home.  Not really worried about this though. 20 min with the screen off is solid. When she wears the watch it will be turning the screen on with notifications and such and that will keep it alive for another 20 min. I created a custom sensor in home assistant that updates her location with the latest device that updated location (watch vs iphone).

**Comment by RacoJo** [comment]
Hello, i got the same problem. Iphone is tracking perfectly (with IRK) only when screen is on (even without unlocking it work fine, but screen has to be on).
Is there a solution for this ?

**Comment by lxgenie** [comment]
Just started using ESPresense and also seeing the issue where tracking is only happening when the screen is on. I'm using the IRK as the id to track.

Is there any further solution to this?

**Comment by bayees** [comment]
I have two iPhones in the household and one is working perfectly the other one stops while locked. I am experimenting with the Covid exposesure app and room-assistant app right now but no conclusion.

**Reply by lxgenie** [reply]
Have you compared settings on both phones to see if there is any different?

Are they both the same model and running same version of iOS?

**Comment by huytd2k** [comment]
After installing the room-assistant app here: https://apps.apple.com/in/app/room-assistant/id1538642237, The issue seems to be fixed for me

**Comment by Terastar-Paperclip** [comment, marked as answer]
This is the #1 case in our canonical "away detection" writeup — Apple devices drop their advertising rate and rotate their BLE address once the screen locks, and the fix is enrolling the iPhone's IRK plus raising `forget_ms` (default 150s — try 300s). Full breakdown, including the "paired watch suppresses phone" variant a few of you hit above: #2322. Marking this answered off that pin since it's exactly the case it was built to cover — reply here if it doesn't fix it for you.
